test/e2e/e2e_test.go

```diff
@@ -169,16 +171,66 @@ var _ = ginkgo.ReportAfterSuite("Kubernetes e2e suite report", func(report ginkg
 			klog.Errorf("Error in detail reporter: %v", err)
 			return
 		}
-		_, err = f.Write(b)
-		if err != nil {
+		if _, err = f.Write(b); err != nil {
 			klog.Errorf("Error saving test details in detail reporter: %v", err)
 			return
 		}
 		// Printing newline between records for easier viewing in various tools.
-		_, err = fmt.Fprintln(f, "")
-		if err != nil {
+		if _, err = fmt.Fprintln(f, ""); err != nil {
 			klog.Errorf("Error saving test details in detail reporter: %v", err)
 			return
 		}
 	}
+}
+
+// writeJUnitReport generates a JUnit file in the e2e report directory that is
+// shorter than the one normally written by `ginkgo --junit-report`. This is
+// needed because the full report can become too large for tools like Spyglass
+// (https://github.com/kubernetes/kubernetes/issues/111510).
+//
+// Users who want the full report can use `--junit-report`.
+func writeJUnitReport(report ginkgo.Report) {
+	if framework.TestContext.ReportDir == "" {
+		return
+	}
+
+	trimmedReport := report
+	trimmedReport.SpecReports = nil
+	for _, specReport := range report.SpecReports {
+		// Remove details for any spec that hasn't failed. In Prow,
+		// the test output captured in build-log.txt has all of this
+		// information, so we don't need it in the XML.
+		if specReport.State != types.SpecStateFailed {
+			specReport.CapturedGinkgoWriterOutput = ""
+			specReport.CapturedStdOutErr = ""
+		}
+
+		// Remove report entries generated by ginkgo.By("doing
+		// something") because those are not useful (just have the
+		// start time) and cause Spyglass to show an additional "open
+		// stdout" button with a summary of the steps, which usually
```

Comment:
we'd better to focus on issue here,  if we all agree that a stdout for a failure testcase are useless, we can address that in a separate pr, WDYT?

Reply:
Let's discuss this issue in another pr, I'd prefer to not touch this in this one.

Reply:
some test use them to log the steps of the test, I think we should keep them and merge, I think that was the previous behavior , but that is for another PR

Reply:
+1

Reply:
The text messages from `ginkgo.By` still shows up in in the output even when trimming report entries. That is independent from each other because `ginkgo.By` now does two things:
- log the output
- create a report entry

I can create a separate PR for trimming the report entries.